staticd/static_nb_config.c

```diff
@@ -35,17 +35,33 @@ static int static_path_list_create(struct nb_cb_create_args *args)
 {
 	struct route_node *rn;
 	struct static_path *pn;
+	const struct lyd_node *vrf_dnode;
+	const char *vrf;
 	uint8_t distance;
+	uint32_t table_id;
 
 	switch (args->event) {
 	case NB_EV_VALIDATE:
+		vrf_dnode = yang_dnode_get_parent(args->dnode,
+						  "control-plane-protocol");
+		vrf = yang_dnode_get_string(vrf_dnode, "./vrf");
+		table_id = yang_dnode_get_uint32(args->dnode, "./table-id");
+		if (table_id && (strcmp(vrf, vrf_get_default_name()) != 0)
+		    && !vrf_is_backend_netns()) {
+			snprintf(
+				args->errmsg, args->errmsg_len,
+				"%% table param only available when running on netns-based vrfs");
```

Comment:
hmm - I think specifying the table for a route has been supported. is this new - is there a reason to impose this restriction?

Reply:
yea this shouldn't be restricted. vrf_netns or vrf-lite, people should be allowed to install routes to whatever tables they want with staticd.

Reply:
In the context of below PR, we had a discussion and we agreed on above check. 
[(https://github.com/FRRouting/frr/pull/7329)]

Reply:
Oh is this an instance where we have both VRF and table ID?  You may be right to disallow then.


Specifically, a user should be able to do `ip route 1.1.1.1/32 2.2.2.2 table 1001` regardless of vrf-netns or vrf-lite.

as long as they can still do the above, I am fine with it.

Reply:
maybe it would be clearer if the ordering of the tests in that block were different: test for the presence of a vrf first then test for a table_id and not-netns, to identify the invalid case?

Reply:
@mjstapp :  We need to test this only, when there is a valid tableId(Here valid means > 0).
This API gets called even when tableId is zero. So It's better we should check tableId first.
I the other case we keep on executing vrf check, even when tableId is zero.

Reply:
so ... it sounds as if you think there's a performance problem - but of course, since this is config and yang and nb and we know that is very slow, it's hard to imagine that making the intent of this block clearer is really going to have a performance impact?

if you don't want to change the order, then we need a comment - since two of us had trouble decoding the intent of the tests? "table id and vrf together allowed only for netns vrf backend"

Reply:
Sure Mark, I will add the comment. 
Yang and NB are slow, agreed,  But we are doing our best to achieve scale.

Reply:
Done, Thanks @mjstapp and @sworleys for your review.